### ISSUE TITLE
[VPAT] Text selection widgets for VPAT

### ISSUE THREAD

**Issue by Renzo-Olivares** [flutter/flutter#185220]
Tracking issue for VPAT assesment of Flutter's text-selection components. These widgets should have their own dedicate use case screen in https://github.com/flutter/flutter/tree/master/dev/a11y_assessments (checked widgets have already been added).

### Core Selection

These widgets are primary targets for accessibility as they hold the core logic for text selection manipulation which includes keyboard shorcut handling, and overlays like the selection context menu, selection handles, and text magnifier. They need to be able to clearly communicate their selection state to the accessibility system.

- [ ] SelectionArea
- [ ] SelectableRegion
- [ ] SelectableText
